src/main/java/org/folio/circulation/services/AllowedServicePointsService.java

```diff
@@ -83,6 +89,22 @@ private CompletableFuture<Result<RequestPolicy>> fetchItemAndLookupRequestPolicy
       .thenCompose(r -> r.after(item -> lookupRequestPolicy(item, user, request)));
   }
 
+  private CompletableFuture<Result<Set<RequestPolicy>>> fetchItemsByInstanceAndLookupRequestPolicies(
+    AllowedServicePointsRequest request, User user) {
+
+    log.debug("lookupRequestPolicy:: parameters request: {}, user: {}", request, user);
+
+    if (user == null) {
```

Comment:
We fetch the user for both ILR and TLR requests. Can we fetch it once and validate the result at the very top in `getAllowedServicePoints()` ?

Reply:
done